src/rgw/rgw_rest_pubsub.cc

```diff
@@ -195,13 +195,17 @@ class RGWPSCreateTopicOp : public RGWOp {
       return 0;
     }
     if (ret == 0) {
-      if (result.user == s->owner.id ||
-          !s->cct->_conf->mandatory_topic_permissions) {
+      ret = verify_topic_owner_or_policy(
```

Comment:
`snsCreateTopic` was not added as policy coz it wad decided that `CreateTopic` can be only called by the existing user, so calling `verify_topic_owner_or_policy` here is a `NO_OP`.
unless we add the `snsCreateTopic` as iam policy, to achieve backward compatibility we could just check for user.empty or its the owner here instead of calling `verify_topic_owner_or_policy`

Reply:
`sns:CreateTopic` does not exists in AWS SDKs.
also, i'm not sure why we made that decision. why is `Createtopic` different than `SetTopicAttributes`?

Reply:
> `sns:CreateTopic` does not exists in AWS SDKs.
> also, i'm not sure why we made that decision. why is `Createtopic` different than `SetTopicAttributes`?

SetAttribute is completely different than CreateTopic. Createatopic can be called without setting any attributes. 
Since original code used same Createatopic() for both function we think it's the same. However we do not modify any param for SetAtribute other than one passed, whereas in createtopic we overwrite all the params whether passed or not

Reply:
the only mention of permissions in https://docs.aws.amazon.com/sns/latest/api/API_CreateTopic.html#API_CreateTopic_RequestParameters is a note under Tags:
> To be able to tag a topic on creation, you must have the sns:CreateTopic and sns:TagResource permissions.

we don't support tags, and it doesn't say that CreateTopic needs sns:SetTopicAttributes to add Attributes, so we should just require sns:CreateTopic by default

the owner doesn't need special permission to recreate, but they could use topic policy to grant that permission to other users

Reply:
ok. that was quite hidden...
will chaneg the code to check for sns:CreateTopic